producer/writer/consumer_writer.go

```diff
@@ -200,9 +200,16 @@ func (w *consumerWriterImpl) Init() {
 	}()
 }
 
-func (w *consumerWriterImpl) resetConnectionUntilClose() {
+func (w *consumerWriterImpl) cleanupConnectionUntilClose() {
+	flushTicker := time.NewTicker(w.connOpts.FlushInterval())
+	defer flushTicker.Stop()
+
 	for {
 		select {
+		case <-flushTicker.C:
```

Comment:
Any perf concerns because of the more frequent flush?

Reply:
This actually does less frequent but larger flush, because now we can use larger buffer size, so during the beginning of the minute, more messages can be buffered into one flush, this ticker only ticks once about 100ms or 1s'ish to limit the max delay

Reply:
Did you see visible perf improvements due to larger flushes? I can see some perf gains due to writing less with bigger batches, but I'm kinda surprised that writing 16K at a time vs writing 1K at a time makes that big of a difference.

Reply:
Yeah, I remember back when I was benchmarking this, basically the larger the buffer size is faster the speed, 16K is probably 3-4X or more faster than 1K, in fact grpc use 32K by default, we could even use larger buffer like 32k or 64k ourselves.

Reply:
Hm interesting, I would imagine that there's some benefit in doing fewer `write` calls in the userspace and kernel space but the kernel would need to chunk the 16K buffers into ~1K packets due to MTU limitation and reassemble them on the receiver side, which offsets some of the benefit. Perhaps the reduced call overhead significantly outweighs the packet chunking+assembling overhead, but still somewhat surprised. I'd be interested to see what the perf difference is when you run this in staging and production.